src/fidgets/farcaster/components/CreateCast.tsx

```diff
@@ -0,0 +1,297 @@
+    }
+
+    if (draft?.embeds) {
+      setInitialEmbeds(draft.embeds);
+    }
+  }, [editor]);
+
+  const text = getText();
+  const embeds = getEmbeds();
+  const channel = getChannel();
+
+  useEffect(() => {
+    if (!editor) return; // no updates before editor is initialized
+    if (isPublishing) return;
+    if (draft?.parentUrl === channel?.parent_url) return;
+
+    const newEmbeds = initialEmbeds ? [...embeds, ...initialEmbeds] : embeds;
+    setDraft({
+      ...draft,
+      text,
+      embeds: newEmbeds,
+      parentUrl: channel?.parent_url || undefined,
+    });
+  }, [text, embeds, initialEmbeds, channel, isPublishing, editor]);
+
+  const getButtonText = () => {
+    if (isLoadingSigner) return "Not signed into Farcaster";
+    if (isPublishing) return "Publishing...";
+    return "Cast";
+  };
+
+  return (
+    <div
+      className="flex flex-col items-start min-w-full w-full h-full"
+      tabIndex={-1}
+    >
+      <form onSubmit={handleSubmit} className="w-full">
+        {isPublishing ? (
+          <div className="w-full h-full min-h-[150px]">
+            <Skeleton className="px-2 py-1 w-full h-full min-h-[150px] text-foreground/80">
+              {draft.text}
+            </Skeleton>
+          </div>
+        ) : (
+          <div className="p-2 border-slate-200 rounded-lg border">
+            <EditorContent
+              editor={editor}
+              autoFocus
+              className="w-full h-full min-h-[150px] text-foreground/80"
+            />
+            <EmbedsEditor
+              embeds={[]}
+              setEmbeds={setEmbeds}
+              RichEmbed={() => <div />}
+            />
+          </div>
+        )}
+
+        <div className="flex flex-row pt-2 gap-1">
+          {!isReply && (
+            <div className="text-foreground/80">
+              {isPublishing || isLoadingSigner ? (
+                getChannel().name
+              ) : (
+                <ChannelPicker
+                  getChannels={debouncedGetChannels}
+                  onSelect={setChannel}
+                  value={getChannel()}
+                />
+              )}
+            </div>
+          )}
+          <Button
+            className="h-10"
+            type="button"
+            variant="outline"
+            disabled={isPublishing}
+            onClick={() => setCurrentMod(creationMods[0])}
+          >
+            <PhotoIcon className="mr-1 w-5 h-5" />
+            Add
+          </Button>
+          <Popover
+            open={!!currentMod}
+            onOpenChange={(op: boolean) => {
+              if (!op) setCurrentMod(null);
+            }}
+          >
+            <PopoverTrigger></PopoverTrigger>
+            <PopoverContent className="w-[300px]">
+              <div className="space-y-4">
+                <h4 className="font-medium leading-none">{currentMod?.name}</h4>
+                <hr />
+                <CreationMod
+                  input={getText()}
+                  embeds={getEmbeds()}
```

Comment:
Use the value of `const embeds = getEmbeds();` defined above

Reply:
Fixed all of these. More things copied from herocast